src/config/ConfigManager.cpp

```diff
@@ -1324,6 +1343,8 @@ std::string CConfigManager::parseKeyword(const std::string& COMMAND, const std::
         handleSource(COMMAND, VALUE);
     else if (COMMAND == "submap")
         handleSubmap(COMMAND, VALUE);
+    else if ((COMMAND == "persist" || COMMAND == "reset" || COMMAND == "consume") && m_soCurrentSubmap.getName() != "")
```

Comment:
I'd move the check for submap inside the handler and pop an error there if thats the case